src/java.base/share/classes/java/nio/channels/spi/SelectorProvider.java

```diff
@@ -117,15 +110,7 @@ private static SelectorProvider loadProviderAsService() {
                                    ClassLoader.getSystemClassLoader());
             Iterator<SelectorProvider> i = sl.iterator();
             for (;;) {
-                try {
-                    return i.hasNext() ? i.next() : null;
-                } catch (ServiceConfigurationError sce) {
-                    if (sce.getCause() instanceof SecurityException) {
-                        // Ignore the security exception, try the next provider
-                        continue;
-                    }
-                    throw sce;
-                }
+                return i.hasNext() ? i.next() : null;
```

Comment:
Same thing as AsynchronousChannelProvider here, the loop goes away, it can be replaced with `return sl.findFirst().orElse(null);`